File: src/main/java/com/wanderers/hotelier_webservices/server/exception/BookingServiceException.java
package com.wanderers.hotelier_webservices.server.exception;

import com.wanderers.hotelier_webservices.server.service.impl.BookingServiceImpl;

/**
 * Custom Exception class which used to throw failures in {@link BookingServiceImpl}
 */
public class BookingServiceException extends ServerException {

    private static final long serialVersionUID = 1L;

    /**
     * Class constructor
     *
     * @param message String value of Exception
     */
    public BookingServiceException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/exception/BookingServiceException.kt
package com.wanderers.hotelier_webservices.server.exception

import com.wanderers.hotelier_webservices.server.service.impl.BookingServiceImpl

/**
 * Custom Exception class which used to throw failures in [BookingServiceImpl]
 */
class BookingServiceException
/**
 * Class constructor
 *
 * @param message String value of Exception
 */
    (message: String?, cause: Throwable?) :
    ServerException(message, cause) {
    companion object {
        private const val serialVersionUID = 1L
    }
}